python/pyarrow/tests/test_types.py

```diff
@@ -1331,10 +1331,13 @@ def __init__(self, schema):
         def __arrow_c_schema__(self):
             return self.schema.__arrow_c_schema__()
 
-    schema = pa.schema([pa.field("field_name", pa.int32())])
+    schema = pa.schema([pa.field("field_name", pa.int32())], metadata={"a": "b"})
+    assert schema.metadata == {b"a": b"b"}
     wrapped_schema = Wrapper(schema)
 
     assert pa.schema(wrapped_schema) == schema
+    assert pa.schema(wrapped_schema).metadata == {b"a": b"b"}
```

Comment:
Added this additional line, because `assert pa.schema(wrapped_schema) == schema` (the line above) in itself does not check the metadata (that gets ignored in basic equality testing)